org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/indexing/AddFolderToIndex.java

```diff
@@ -103,8 +104,7 @@ public boolean visit(IResourceProxy proxy) /* throws CoreException */{
 			}
 		} catch (CoreException e) {
 			if (JobManager.VERBOSE) {
-				Util.verbose("-> failed to add " + this.folderPath + " to index because of the following exception:", System.err); //$NON-NLS-1$ //$NON-NLS-2$
-				e.printStackTrace();
+				JavaModelManager.getTrace().trace(null, "-> failed to add " + this.folderPath + " to index because of the following exception:", e); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
I would add `static Util.trace()` method that can redirect to  `JavaModelManager.getTrace().trace()` and so we could simply write here instead:
`trace(null, "-> failed to add " + this.folderPath + " to index because of the following exception:", e); //$NON-NLS-1$ //$NON-NLS-2$`

Reply:
I tried that in my first attempt and the result was that the log trace always showed the line inside `org.eclipse.jdt.internal.core.JavaModelManager | trace` as the source instead of the "real" source _i.e._ this:

```java
| Java indexing | 2023-09-13 16:44:01.842 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.JavaModelManager | trace | 5722 |  | java.lang.RuntimeException: This should be logged into the log file
```

... instead of this:
```java
| Java indexing | 2023-09-13 16:43:46.535 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.search.indexing.SourceIndexer | indexDocument | 119 |  | java.lang.RuntimeException: This should be logged into the log file
```

Reply:
OK, this would mean to have trace() method in every class that uses tracing to have clear code and clear log? Hmm. Not sure what would be the best compromise.

Reply:
But obviously the main goal is to have readable debug trace. That would mean no utility methods...

Reply:
Ach, it looks like Christmas came early this year 🎅 

There's a way to have it both ways: I simply need to use the other constructor and specify `JavaModelManager.class` as the 2nd parameter (`traceEntryClass)` --> `org.eclipse.osgi.service.debug.DebugOptions.newDebugTrace(String, Class<?>)`

I made the appropriate changes in 249c934d99a7685526b565f223d2b15112c4cc25

Reply:
@iloveeclipse does the code look ok now?

Reply:
I'm pretty busy now, may be next week.

Reply:
... ping :-)